### PR TITLE
Remove GoogleUtilities from repo

### PR DESCRIPTION
**Pull request by paulb777**
Removes GoogleUtilities from the repo now that it has been moved to https://github.com/google/GoogleUtilities

Pre-release integration testing with Firebase should now be done via the SpecsStaging repo or path specifications in Podfiles.

#no-changelog

### CODE REVIEW

**Comment by paulb777** [comment]
TODO: figure out cmake rule adjustments to find GoogleUtilities headers and solve failures like https://github.com/firebase/firebase-ios-sdk/runs/1714309783?check_suite_focus=true#step:6:3169

**Comment by paulb777** [comment]
Rebased after #7324 addressed the cmake build transition

File: Example/watchOSSample/Podfile [hunk 1]
```diff
@@ -1,10 +1,12 @@
+source 'https://github.com/firebase/SpecsStaging.git'
```
Comment (ryanwilson) [inline Example/watchOSSample/Podfile:1]: Is this temporary for now or should it stay forever?

**Review by ryanwilson — APPROVED** [review]

File: Example/watchOSSample/Podfile (hunk 1 above)
Comment (paulb777) [reply Example/watchOSSample/Podfile:1]: It should stay forever to support GoogleUtilities integration testing

File: cmake/GoogleUtilities.cmake [hunk 2]
```diff
@@ -34,6 +34,12 @@ firebase_ios_add_framework(
   ${headers} ${sources}
 )
 
+# GoogleUtilities
+target_include_directories(
+  GoogleUtilities PRIVATE
+  $<BUILD_INTERFACE:${FIREBASE_EXTERNAL_SOURCE_DIR}/GoogleUtilities>
```
Comment (wilhuff) [inline cmake/GoogleUtilities.cmake:40]: `PRIVATE` properties only apply at build time anyway, so using `$<BUILD_INTERFACE:...>` is redundant.

**Pull request merged**